### PR TITLE
IRLock: added selection of I2c bus

### PR DESCRIPTION
**Pull request by tridge**
this is useful on Pixhawk2

### CODE REVIEW

**Comment by tridge** [comment]
this is pending testing by Leonard

**Comment by rmackay9** [comment]
In the hopes of easing set-up and keeping the number of parameters low, I wonder if it would be possible to automatically detect whether the sensor is there or not and then try each bus?

**Comment by rmackay9** [comment]
@tridge, yeah, not surprised.  The protocol is a bit odd.  It as if someone decided that implementing serial over I2C was a good idea.

File: libraries/AP_IRLock/AP_IRLock_SITL.h [hunk 1]
```diff
@@ -16,7 +16,7 @@ class AP_IRLock_SITL : public IRLock
     AP_IRLock_SITL();
 
     // init - initialize sensor library
-    virtual void init();
+    virtual void init(int8_t bus);
```
Comment (OXINARF) [inline libraries/AP_IRLock/AP_IRLock_SITL.h]: `override` please

File: libraries/AC_PrecLand/AC_PrecLand.cpp [hunk 2]
```diff
@@ -83,6 +83,13 @@ const AP_Param::GroupInfo AC_PrecLand::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("CAM_POS", 7, AC_PrecLand, _cam_offset, 0.0f),
 
+    // @Param: BUS
+    // @DisplayName: Sensor Bus
+    // @Description: Precland sensor bus for I2C sensors.
+    // @Values: -1:DefaultBus,0:InternalI2C,1:ExternalI2C
```
Comment (OXINARF) [inline libraries/AC_PrecLand/AC_PrecLand.cpp]: Any reason the value descriptions don't have spaces? Harder to read.

**Comment by lthall** [comment]
I tested it and it looks good. The only thing that may be an issue is the default I2C port for the cube. I am not sure which one it should be. I had it on I2C-2.

**Comment by OXINARF** [comment]
@tridge Am I missing something or are we just doing reads? Or did I got the transfer method the wrong way?

**Comment by Kiwa21** [comment]
@rmackay9 I was using the following code with success on the Cube I2C_2 connector without adding a new param :  

AC_IRLock_I2C.cpp
`
dev = std::move(hal.i2c_mgr->get_device(0, IRLOCK_I2C_ADDRESS));
if(!dev){
	dev = std::move(hal.i2c_mgr->get_device(1, IRLOCK_I2C_ADDRESS));
}
`

**Comment by OXINARF** [comment]
@Kiwa21 that doesn't work. It is working for the port you are using because it is the first the code is trying.

File: libraries/AP_IRLock/AP_IRLock_SITL.h (hunk 1 above)
Comment (tridge) [inline libraries/AP_IRLock/AP_IRLock_SITL.h]: fixed, thanks!

**Comment by tridge** [comment]
@OXINARF you're right, it is only reads, I mis-read it (sorry for the pun!)
it still won't auto-detect though. I remember now the problem. It only sends data when it sees the target object. So we can't wait for data at boot.

**Pull request merged**